src/main/java/com/esri/core/geometry/Envelope.java

```diff
@@ -1030,7 +1010,7 @@ public boolean equals(Object _other) {
 	public int hashCode() {
 		int hashCode = m_description.hashCode();
 		hashCode = NumberUtils.hash(hashCode, m_envelope.hashCode());
-		if (!isEmpty() && m_attributes != null) {
+		if (!isEmpty()) {
```

Comment:
Are we sure at this point that m_attributes != null?

Reply:
There should not be a problem here. I am making sure that m_attributtes are not null if the geometry has any extra attributes (z, m, etc).